docs/architecture-decisions/2024-01-31_binary_distribution.md

```diff
@@ -0,0 +1,254 @@
+- Codesigning is a recommended practice for secure distribution and would need to be implemented on top of the initial workflows introduced as part of the implementation of the [native binaries](./2024-01-13_native_binaries.md) ADR.
+
+## Options
+
+### Option 1 - Binaries are only available via dedicated package managers using OSS solutions for multi architecture support
+
+This approach assumes that native binaries are never available for direct consumption but instead extra tailored per each selected package manager. Additionally this approach assumes using OSS solutions for multi architecture support. Primarily using a QEMU/buildx based actions to build the binaries for different linux architecutres, using paid m1 github runner for arm macos binaries, using default macos runner for x86 macos binaries and using default windows runner for x86-64 windows binaries.
+
+Diagram:
+
+```mermaid
+flowchart
+    n1["Release pipeline flow"]
+	n1 --- n2["Python wheel build &amp; test validation (runs on [3.10, 3.11, 3.12 on windows, mac, ubuntu]"]
+	n2 --- n3["Build binaries"]
+	n3 --- n4["Linux"]
+	n3 --- n5["Mac"]
+	n3 --- n6["Windows"]
+	n4 --- n7["QEMU aarch64-ubuntu22.04"]
+	n4 --- n8["QEMU armv7-ubuntu22.04"]
+	n7 --- n9["Upload artifacts"]
+	n5 --- n10["macos-latest x86-64"]
+	n5 --- n11["macos-large-latest ARM (paid worker)"]
+	n6 --- n12["windows-latest (x86-64)"]
+	n8 --- n9
+	n10 --- n9
+	n11 --- n9
+	n12 --- n9
+	n9 --- n13["Distribution"]
+	n13 --- n14["pypi pure python wheel"]
+	n13 --- n15["snap"]
+	n13 --- n16["brew"]
+	n13 --- n17["winget"]
+	n17 --- n18["refresh manifest"]
+	n18 --- n19["submit release PR to winget community repo "]
+	n16 --- n20["refresh cask definition (can use conditional that auto picks either ARM or Intel based artifacts)"]
+	n20 --- n21["submit pr to algorandfoundation/tap"]
+	n15 --- n22["aarch64"]
+	n15 --- n23["armv7"]
+	n22 --- n24["submit to Launchpad to build snap for aarch64 on amd64 architecture"]
+	n23 --- n25["submit to Launchpad to build snap for armv7 binary on arm64 architecture"]
+	n24 --- n26["publish snap"]
+	n25 --- n26
+```
+
+#### General pros and cons
+
+**Pros**
+
+- Ability to rely on package manager to handle installation, updates and removal of the binaries
+- OSS solutions for multi architecture support are available and can be used to build the binaries for different architectures on Linux using QEMU
+
+**Cons**
+
+- Increase complexity of release pipeline as we need to introduce additional steps to build the binaries for different architectures as well as introduce tailored workflows for each package manager
+- PoC tests on building pyinstaller inside QEMU armv7 ubuntu images are proven to be **extremely** slow (up to an hour without cached based docker images). Additionally the build process is complicated by the fact that wheels like `cryptography` require `rust` to be installed as part of building the wheel, which increases the build time even further.
```

Comment:
We should note that this only applies to ARM builds or non-Ubuntu distros, both of which for now we have decided not to support.